pkg/api/controller.go

```diff
@@ -170,6 +170,11 @@ func (c *Controller) Run(reloadCtx context.Context) error {
 		return err
 	}
 
+	if c.Config.HTTP.Port == "0" || c.Config.HTTP.Port == "" {
```

Comment:
right side of the condition is never true because zot will have the port 8080 as default